spec/request/v0/facilities/ccp_request_spec.rb

```diff
@@ -250,8 +302,8 @@
           'meta' => {
             'pagination' => {
               'current_page' => 1,
-              'per_page' => 20,
-              'total_pages' => 1,
+              'per_page' => 10,
+              'total_pages' => 2,
               'total_entries' => 10
```

Comment:
This is illustrative of my questions from above:
* we are on page 1
* there are 10 results per page
* there are 10 results total

Shouldn't `total_pages` be 1?

Reply:
- We are on Page 1
- We have no idea how many pages there are
- if we leave it as it is we will never request the next set of results.

Reply:
My first thought was "Why do we have no idea how many pages there are?"
And then I went down the rabbit hole.
[Here](https://github.com/department-of-veterans-affairs/vets-api/blob/a5cf56e9ace70ab64c38acba41c8f87e0b16845c/lib/facilities/ppms/client.rb#L169) is why we don't know the total number of results:
`maxResults: 20 * page + 1`

We're requesting a different number of results based on what page the end user is on.

The logic for this ([PR](https://github.com/department-of-veterans-affairs/vets-api/pull/2595)) was
```
Tweaks the number of results to work around performance issues with the provider locator ppms endpoint identified in the staging environment. Results now limited to the amount necessary to display the current page (+1 so that a next page will show up in the front end if appropriate)
```

So, given all that, the logic for `total_pages` should be something like:
```ruby
total_pages = if ppms_results.length == per_page * page + 1 
  current_page + 1
else
  current_page
end
```
This should prevent the frontend from displaying a "next page" link when there is no next page.

We also need to ensure that `maxResults` is using the `per_page` value as determined in the controller instead of being hard coded as `20`.

Reply:
And even with all that logic, the service responses can be weird.  I was playing around in a console:
```ruby
ppms = Facilities::PPMS::Client.new
params ={:address=>"366 Main Street, Waterville, Maine 04901, United States", :bbox=>["-70.39", "43.82", "-68.89", "45.32"], :type=>"cc_provider", :services=>["204D00000X"], :page=>"1", :per_page=>"20"}
providers = ppms.provider_locator(params)
providers.count
# => 21.
# OK, that means there is page 2 data.
params[:page] = "2"
providers = ppms.provider_locator(params)
providers.count
# => 32
# OK, that means we've retrieved all the results.  We would have received a count of 41 if there were page 3 results.  But maybe I'll try page 3 anyway
params[:page] = "3"
providers = ppms.provider_locator(params)
providers.count
# => 37
# WUT!?!?
```